activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -616,7 +624,15 @@ def remove_columns(table_name, *column_names, **options)
       # to provide these in a migration's +change+ method so it can be reverted.
       # In that case, +type+ and +options+ will be used by #add_column.
       # Indexes on the column are automatically removed.
+      #
+      # If the options provided include an +if_exists+ key, it will be used to check if the
+      # column does not exist. This will silently ignore the migration rather than raising
+      # if the column was already used.
+      #
+      #   remove_column(:suppliers, :qualification, if_exists: true)
       def remove_column(table_name, column_name, type = nil, **options)
+        return if options[:if_exists] == true && !column_exists?(table_name, column_name)
```

Comment:
Should we pass the type into the exists check here too?

Reply:
I thought about this and tried it out and there's an issue with passing type. For `add_column` the type is required, you have to pass it so we can use that to check if the `column_exists?`.

However, in `remove_column` it's allowed but completely ignored by the actual column dropper. For `remove_column` since type is ignored by the execute we don't need to pass `type` there because the column will be removed regardless of type so we don't want to protect on nil or type. Does that make sense? I feel like I'm rambling...

Reply:
Basically doesn't matter what you pass for type, remove column will always remove the column because type is ignored.

Reply:
Got it, perfect 👌